### PR TITLE
EKS Cluster Log collector script

### PR DESCRIPTION
**Pull request by milanoo7**
This is EKS- Cluster Log collector script , which will capture EKS POD logs and other K8 resources details and create tarball that Cx can share on their EKS support case for efficient troubleshooting. 

NOTE: Make sure to run the script from your local machine against your EKS cluster with Admin RBAC permission. It will create a captured log folder in your present directory.

Run command: ./EKS-ClusterCrawler.sh <namespace name>

**Issue #, if available:**

**Description of changes:**
 Already provided working Demo to "Todd Neal" and upon his suggestion creating this PR request for further security review before we publish it for customer.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Testing Done**

*[See this guide for recommended testing for PRs.](../doc/CONTRIBUTING.md#testing-changes) Some tests may not apply. Completing tests and providing additional validation steps are not required, but it is recommended and may reduce review time and time to merge.*

### CODE REVIEW

File: log-collector-script/EKS-ClusterCrawler.sh [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+# echo "Collecting recent events log that took place within ${CLUSTERNAME}, Review file: EventsInfo.txt"
+# FILENAME5="${OUTPUT_DIR}/EventsInfo.txt"
+# kubectl get events -A --sort-by=.metadata.creationTimestamp > "$FILENAME5"
+
+# # Collecting All the Worker-Node level descriptions and their performance metrics informantion.
+# echo "Collecting information about all the presently running worker node within ${CLUSTERNAME}, Review file: WorkerNodeInfo.txt "
+# FILENAME6="${OUTPUT_DIR}/WorkerNodeInfo.txt"
+# kubectl describe node -A > "$FILENAME6"
+
+# # Collecting All the information about persistance volume and storage class.
```
Comment (ravisinha0506) [inline log-collector-script/EKS-ClusterCrawler.sh:75]: We should remove this commented logic from here.

File: log-collector-script/EKS-ClusterCrawler.sh [hunk 2]
```diff
@@ -0,0 +1,129 @@
+# #!/bin/bash
```
Comment (ravisinha0506) [inline log-collector-script/EKS-ClusterCrawler.sh:1]: I am not sure if this script should be reside inside an AMI?

File: log-collector-script/EKS-ClusterCrawler.sh [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+kubectl get ns --no-headers | while read -r line; do
+NAMESPACE=$(echo "$line" | awk '{print $1}')
+if [[ "$NAMESPACE" = "$Default_Namespace" ]];then
+    kubectl get pods -n "$NAMESPACE" --no-headers | while read -r lines; do
+            POD_NAME=$(echo "$lines" | awk '{print $1}')
+            echo "$POD_NAME" 
+            FILENAME1="${OUTPUT_DIR}/${NAMESPACE}.${POD_NAME}.describe.txt"
+            kubectl describe pod -n "$NAMESPACE" "$POD_NAME" > "$FILENAME1"
+            for CONTAINER in $(kubectl get po -n "$NAMESPACE" "$POD_NAME" -o jsonpath="{.spec.containers[*].name}"); do
+            FILENAME_PREFIX="${OUTPUT_DIR}/${NAMESPACE}.${POD_NAME}.${CONTAINER}"
```
Comment (ravisinha0506) [inline log-collector-script/EKS-ClusterCrawler.sh:39]: We are looking each and every container in the cluster. Have we load tested this script?

File: log-collector-script/EKS-ClusterCrawler.sh (hunk 2 above)
Comment (cartermckinnon) [reply log-collector-script/EKS-ClusterCrawler.sh:1]: Agreed, we've looked at a similar script previously: https://github.com/awslabs/amazon-eks-ami/pull/886#issuecomment-1324119315

This type of resource should go in a dedicated troubleshooting repository, not the AMI repo.

**Review by cartermckinnon — CHANGES_REQUESTED** [review]

**Pull request closed without merging**